### PR TITLE
Feature: Link to User Profiling Survey at the end of the Tutorial Feedback Survey

### LINKED ISSUE #424: User profile survey
After implementing the 3-question on-screen tutorial satisfaction survey (#423), we plan to create a longer survey in Google Forms designed to learn more about the profile of our typical user. Users who've completed the tutorial feedback will be asked if they can spare the time to fill this one out. Data collected will include questions such as whether a user is a developer, how new they are to coding and to dweb, how they learned about ProtoSchool, etc. 

Blocked by #423, which we'll launch on its own and then adapt to lead optionally to this longer survey. 

Once you've completed this longer survey, we'll cache your status so you won't see the request a second time unless visiting in a new browser or otherwise clearing your cache. 

Planning notes at: https://docs.google.com/document/d/1z-QA-bV-6YnZW0HQgvLNLhEwFakgsjls8inYB0gZtCE/edit?usp=sharing

Will share a Google Forms draft when ready.

### PR DESCRIPTION
**Pull request by zebateira**
image

### TODO

- [x] Real survey link (ATM is `https://google.com`)

#### Implementation details
For this we had to split the button into two components: Button and
ButtonLink.

The normal Button component should be used for situations when we
need to control the button click action.
The ButtonLink is simply an anchor tag with the look and feel of the
button with the new option `external` to be an external link.

Closes #424

### CODE REVIEW

**Comment by terichadbourne** [comment]
@zebateira Do we need some conditional logic for displaying this? I'm imagining we might want to cache status when you click to go to the profile survey and then never show the request/link again, instead showing the version we have now. Does that make sense to you?  We may lose some people who click out to it but don't complete it and never see the  request again, but I think the only other option is to show it forever even if you've done it.

**Comment by terichadbourne** [comment]
@zebateira My latest commit adds logic and conditional display of either the old ThankYouMessage message or the new one (with link) based on whether or not you've already completed the survey. I tried to use you settings helpers to adjust localStorage but didn't find the documentation clear enough to follow. Maybe you could walk me through it tomorrow? We should also check that the design hasn't been negatively affected by using the component in two ways and figure out how we want to make the thank you with prompt disappear after the button's been clicked. My Vue thingy in the inspector isn't working at the moment so I was flying blind a bit here. Let me know what you'd suggest for improvements!

**Comment by zebateira** [comment]
@terichadbourne so the only thing you were missing was [the export of the profileSurvey object](https://github.com/ProtoSchool/protoschool.github.io/pull/451/files#diff-74a25d27897129af64a30b5549482056R101).
I'll add more docs on how to use the module to make it more clear.

**Comment by zebateira** [comment]
Let's also add to the tracking data of the `dismiss` event if the user us dismissing the profile survey link or not.

**Comment by terichadbourne** [comment]
[Survey in Google Forms](https://forms.gle/yPXid3EpNQkrpDqM7) is now available for review. Please note **all entries are currently for test purposes and will be deleted before we merge this PR** (so don't hesitate to hit submit and get the full experience while testing).

File: src/static/translations/en.json [hunk 1]
```diff
@@ -23,8 +23,9 @@
     },
     "thankYouMessage": {
       "title": "Thank you!",
-      "text": "<p>Your feedback is very much appreciated!",
-      "text-complete": "<p>Your feedback is very much appreciated!<br />Would you like to help us out a bit more?</p><p>  If you want, you can fill up this quick survey to helps us improve our content.</p>"
+      "textWithPrompt": "<p class='b mt2 mb0'>Have a few more minutes to help us out?</p><p style='margin-top: 0;'>We've created a five-minute survey to learn more about our users so we can better meet your needs.</p>",
```
Comment (terichadbourne) [inline src/static/translations/en.json:26]:
```suggestion
      "textWithPrompt": "<p class='b mt2 mb0'>Have 3 more minutes to help us out?</p><p style='margin-top: 0;'>We'd love to learn more about you so we can create content and features that meet your needs. Take this anonymous survey to share your experience with ProtoSchool, DWeb, and coding.</p>",
```

File: src/static/translations/en.json [hunk 2]
```diff
@@ -23,8 +23,9 @@
     },
     "thankYouMessage": {
       "title": "Thank you!",
-      "text": "<p>Your feedback is very much appreciated!",
-      "text-complete": "<p>Your feedback is very much appreciated!<br />Would you like to help us out a bit more?</p><p>  If you want, you can fill up this quick survey to helps us improve our content.</p>"
+      "textWithPrompt": "<p class='b mt2 mb0'>Have 3 more minutes to help us out?</p><p style='margin-top: 0;'>We'd love to learn more about you so we can create content and features that meet your needs. Take this anonymous survey to share your experience with ProtoSchool, DWeb, and coding.</p>",
```
Comment (zebateira) [inline src/static/translations/en.json:26]: Not sure if this is more correct:

```suggestion
      "textWithPrompt": "<p class='b mt2 mb0'>Have 3 more minutes to help us out?</p><p style='margin-top: 0;'>We'd love to learn more about you so we can create content and features that meet your needs. Take this anonymous survey to share your experience with ProtoSchool, DWeb and coding.</p>",
```

Comment (terichadbourne) [reply src/static/translations/en.json:26]: We've been using the Oxford commas throughout, so let's leave as is.

**Pull request merged**